### PR TITLE
CP-13499: Provide physical utilisation metric to RRDD

### PR DESCRIPTION
**Pull request by simonjbeaumont**
The XenAPI changes to expose this metric are not yet merged but I have checked that this is exported by looking at the file exposed in the filesystem: `/dev/shm/metrics/xenvmd-<pid>-stats`.

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
Travis failure is because we're now depending on the rrdd-plugin which requires xenctrl which is not available on this opam switch:

```
opam install thin_lvhd_tools --deps-only

=-=- Synchronising pinned packages =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

[thin_lvhd_tools] /home/travis/build/xapi-project/xenvm/ already up-to-date

The following dependencies couldn't be met:
  - thin_lvhd_tools -> xapi-rrdd-plugin -> xapi-libs-transitional -> xenctrl
Your request can't be satisfied:
  - xenctrl is not available because it requires OCaml >= 4.00.0 & < 4.02.0.
```

File: idl/errors.ml [hunk 1]
```diff
@@ -12,6 +12,7 @@
  * GNU Lesser General Public License for more details.
  *)
 open Lwt
+open Log
```
Comment (euanh) [inline idl/errors.ml]: Do you need to open Log globally when it seems only to be used in fatal_error_t?

File: test/test.ml [hunk 2]
```diff
@@ -50,7 +50,7 @@ let vgs_offline =
 let with_xenvmd ?existing_vg ?(cleanup_vg=true) (f : string -> 'a) =
   let with_xenvmd_running loop =
       xenvm [ "set-vg-info"; "--pvpath"; loop; "-S"; "/tmp/xenvmd"; vg; "--local-allocator-path"; "/tmp/xenvm-local-allocator"; "--uri"; "file://local/services/xenvmd/"^vg ] |> ignore_string;
-      file_of_string "test.xenvmd.conf" ("( (listenPort ()) (listenPath (Some \"/tmp/xenvmd\")) (host_allocation_quantum 128) (host_low_water_mark 8) (vg "^vg^") (devices ("^loop^")))");
+      file_of_string "test.xenvmd.conf" ("( (listenPort ()) (listenPath (Some \"/tmp/xenvmd\")) (host_allocation_quantum 128) (host_low_water_mark 8) (vg "^vg^") (devices ("^loop^")) (rrd_ds_owner \"rrd-ds-owner\"))");
```
Comment (euanh) [inline test/test.ml]: Could this string be split across several lines?  That would make merging future changes easier and clearer.

Comment (jonludlam) [inline test/test.ml]: even better: is the type available at this point? we could generate the config from the type and get a build failure if the type changes.

Comment (simonjbeaumont) [inline test/test.ml]: Nice idea. I'll try and serialise (and de-serialise) it using `Sexplib`.

**Comment by simonjbeaumont** [comment]
@euanh @jonludlam: See above extra 3 commits. If you're now happy I'll squash 9a042b3 and ef80229  into e8071c4 and 8f4ef91 respectively.  2963692 uses Sexplib to generate and parse the config rather than using a string to create the file.

**Comment by euanh** [comment]
@simonjbeaumont: Please go ahead and merge.   I think we could make Xenvmd_stats.start prettier, but it is fine for now. :golf:

**Pull request merged**